fix(curriculum): new step for empty title bug (#55730)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/66ad0f178ed5791ed898fe56.md

---
id: 66ad0f178ed5791ed898fe56
title: Step 66
challengeType: 0
dashedName: step-66
---

# --description--

> If you try to add a new task, add a single space in title field, and click on the `Add New Task` button, you will notice a bug.

Reviewer: ```suggestion
If you try to add a new task, write single space in title field, and click on the `Add New Task` button, you will notice a bug.
```

> An empty task will be created even though title field is required. To fix this, you need add an `if` condition checking for a value that would appear empty.

Reviewer: ```suggestion
An empty task will be created even though the title field is required. To fix this, you need to add an `if` condition checking for a value that would appear empty.
```

Author: review comments

> If you try to add a new task, write single space in title field, and click on the `Add New Task` button, you will notice a bug.
> 
> An empty task will be created even though the title field is required. To fix this, you need to add an `if` condition checking for a value that would appear empty.

Reviewer 2: ```suggestion
Finally, there's one bug to fix. If you create a task with only spaces for the title, an empty task will be created even though the title is required.

Fix that by adding a check at the beginning of the function for if `!titleInput.value.trim()`. If there's no title, show an alert with the text `Please provide a title` and `return` after that.
```

With that, you've completed this project.

# --hints--

You should add an `if` statement with the condition `!titleInput.value.trim()` inside `addOrUpdateTask` function.

```js
assert.match(code, /if\s*\(\s*!titleInput\.value\.trim\(\s*\)\s*\)\s*/)
```

You should call the `alert("Please provide a title")` in your `if` block.

```js

> assert.match(code, /if\s*\(\s*!titleInput\.value\.trim\(\s*\)\s*\)\s*\{\s*\s*alert\s*\(\s*('|"|`)Please provide a title\1\s*\)\s*/)

Reviewer 2: ```suggestion
assert.match(code, /if\s*\(\s*!titleInput\.value\.trim\(\s*\)\s*\)\s*\{\s*alert\s*\(\s*('|"|`)Please provide a title\1\s*\)\s*/)
```

```

Your `if` statement should return from the function.

```js

> assert.match(code, /if\s*\(\s*!titleInput\.value\.trim\(\s*\)\s*\)\s*\{\s*\s*alert\s*\(\s*('|"|`)Please provide a title\1\s*\)\s*;?\s*return\s*;?\s*\}/);

Reviewer 2: ```suggestion
assert.match(code, /if\s*\(\s*!titleInput\.value\.trim\(\s*\)\s*\)\s*\{\s*alert\s*\(\s*('|"|`)Please provide a title\1\s*\)\s*;?\s*return\s*;?\s*\}/);
```

---
id: 66ad0f178ed5791ed898fe56
title: Step 66
challengeType: 0
dashedName: step-66
---

# --description--

> If you try to add a new task, add space in title field and then click on the `Add New Task` button, you will notice a bug.

Reviewer 3: ```suggestion
If you try to add a new task, add a single space in title field, and click on the `Add New Task` button, you will notice a bug.
```

> Empty task will be created even though title field is required. To fix this, you need add an `if` condition checking for empty title field value.

Reviewer 3: ```suggestion
An empty task will be created even though title field is required. To fix this, you need add an `if` condition checking for a value that would appear empty.
```

With that, you've completed this project.

# --hints--

You should add an `if` statement with the condition `!titleInput.value.trim()` inside `addOrUpdateTask` function.

```js
assert.match(code, /if\s*\(\s*!titleInput\.value\.trim\(\s*\)\s*\)\s*/)
```

You should call the `alert("Please provide a title")` in your `if` block.

```js

> console.log("here is")
> console.log(code)

Reviewer 3: ```suggestion
```
